spyderlib/restart_app.py

```diff
@@ -129,18 +181,47 @@ def main():
     shell = os.name != 'nt'
 
     # Wait for original process to end before launching the new instance
-    while True:
+    wait_time = 60*5  # Seconds
+    for counter in range(int(wait_time/SLEEP_TIME)):

```

Comment:
What does this `for` loop exactly do? Could you add a comment below

    # Wait for original process ...

to say something about it?